beancount/parser/lexer.l

```diff
@@ -552,43 +491,68 @@ void yylex_initialize(PyObject* file, PyObject* filename, int lineno, const char
     extra->n_line_tokens = 0;
     extra->line = lineno;
     extra->encoding = encoding ? encoding : "utf-8";
-    buffer_init(&extra->str, 1024);
 
     Py_XDECREF(yyget_in(scanner));
     Py_INCREF(file);
     yyrestart((void *)file, scanner);
     yybegin(scanner);
 }
 
```

Comment:
Please add a documentation comment for this function, in particular describing the special error codes it can return.

Reply:
Also please add comment about ownership of returned buffer.